CoffeeMachineStates/src/CoffeeMachineStates.java

```diff
@@ -0,0 +1,247 @@
+import java.util.Scanner;
+
+public class CoffeeMachineStates {
+    int water = 400;
+    int milk = 540;
+    int coffeeBeans = 120;
+    int cups = 9;
+    int money = 550;
+    States state = States.Ready;
+    Scanner scanner = new Scanner(System.in);
+
+    /**
+     * this method is brain for our class
+     * its  works with states and change our machine states depending
+     * on our choice,also calls other methods from this class
+     *
+     */
+    private void action(String  action){
+        boolean isDigit = isDigit(action);
+
+        if(action.equals("buy")){
```

Comment:
@AraChilingaryan jan you can use Enum for action types . Also always compare literal with string object . For example "buy".equals(action) . Because if action will be null in that case null.equals will work right ? which will throw an exception. So to avoid this swap them